client/state/index.js

```diff
@@ -168,7 +166,7 @@ export function createReduxStore( initialState = {} ) {
 		require( './data-layer/wpcom-api-middleware.js' ).default,
 		isBrowser && require( './data-layer/extensions-middleware.js' ).default,
 		noticesMiddleware,
-		isBrowser && require( './happychat/middleware.js' ).default(),
+		isBrowser && require( 'extensions/happychat/state/middleware.js' ).default(),
```

Comment:
The happychat middleware could be broken up into handlers and `installActionHandlers`. This way the middleware code wouldn't have to be referenced from the core Calypso code.

Reply:
I've done that in 69f18e1 but I couldn't complete the transition due to [this bit of code](https://github.com/Automattic/wp-calypso/pull/18163/files#diff-5c95e092eca5072a39e6f71ee21a67b1R57).

As a middleware, we have the ability to listen to every action and do some checks (for example, does it have a `meta.analytics` key?). My understanding of the data-layer is that the only way to assign handlers to actions is by assigning them to a specific `action.type` - am I correct?

Reply:
At the moment, I see two options out of this: 1) let extensions to load its own middleware or 2) make handlers in the data-layer to be attachable by some kind of rule, not only by the `action.type`.